### ISSUE TITLE
Incorrect warning about label shadowing itself

### ISSUE THREAD

**Issue by nstoddard** [rust-lang/rust#25343]
The following code:
```
fn main() {
    || {
        'label: loop {
        }
    };
}

```

Results in this warning:
```
main.rs:3:17: 4:10 warning: label name `'label` shadows a label name that is already in scope
main.rs:3         'label: loop {
main.rs:4         }
main.rs:3:17: 4:10 note: shadowed label `'label` declared here
main.rs:3         'label: loop {
main.rs:4         }
```

Obviously a label can't shadow itself, so there shouldn't be a warning. The issue doesn't occur outside of a closure.

**Comment by mitchmindtree** [comment]
I just came across this one myself, just thought I'd +1
